### ISSUE TITLE
Split region management and navigation

### ISSUE THREAD

**Issue by HerrLoesch** [PrismLibrary/Prism#44]
I know this would be a heavy shift but it might be helpful for futur extensions.

I think that RegionManager currently  does to much things at once and violates the SOLID principles. I see especially the Interface Segregation and Single Responsibility Principles are currently violated because it is, for example, quite easy to add your views directly into the views collection of a region and short-circuit the whole navigation concept. Another example is the complex way to use the navigation journal if you want to have a "navigation back" beahvior. A cleaner way would it be to create a NavigationService like we have in the WinRT version. Instead of using the RegionManager or access the NavigationJournal directly, you can just call these methods on the INavigationService interface.

To summarize the advantages of this change:
-	Interfaces are easier to understand
-	Not that easy to bypass
-	Testing is far easier
-	Same interface can be used for WPF, WinRT and others.
